### PR TITLE
CRM-19626: Add min-amount handling to priceset

### PR DESCRIPTION
**Pull request by jitendrapurohit**
* [CRM-19626: Event Registration page allows registration even if no number \(or the number 0\) has been entered in any ticket quantity boxes](https://issues.civicrm.org/jira/browse/CRM-19626)

### CODE REVIEW

**Comment by monishdeb** [comment]
# UI Review

1. Created price-set that support minimum amount 0
image

2. two Price fields created, one with text numeric and second select field with price-option $0 

3. On Event registration page selected $0 price option
image

4. Was able to register successfully. 
image

**Comment by monishdeb** [comment]
In my opinion this should be reflected in api too with api-test coverage.

**Comment by yashodha** [comment]
merging as per comments

File: CRM/Upgrade/Incremental/sql/4.7.15.mysql.tpl [hunk 1]
```diff
@@ -1,4 +1,7 @@
 {* file to handle db changes in 4.7.15 during upgrade *}
 
 -- CRM-19685 (fix for inconsistencies)
-UPDATE civicrm_contact SET preferred_mail_format = 'Both' WHERE preferred_mail_format IS NULL;
\ No newline at end of file
+UPDATE civicrm_contact SET preferred_mail_format = 'Both' WHERE preferred_mail_format IS NULL;
+
+-- CRM-19626
+ALTER TABLE civicrm_price_set ADD min_amount INT(10) UNSIGNED DEFAULT '0' COMMENT 'Minimum Amount required for this set.';
```
Comment (colemanw) [inline CRM/Upgrade/Incremental/sql/4.7.15.mysql.tpl]: @jitendrapurohit @yashodha Putting schema changes directly into the sql file means that an upgrade will fail if run twice (e.g. in the case of upgrading an RC -> stable). I've added a utility for creating columns (and we could add more such utilities for other small tasks like this) and moved this code here: #9548

**Pull request merged**